controllers/resources/bentorequest_controller.go

```diff
@@ -1082,7 +1082,7 @@ func getBentoImageName(bentoRequest *resourcesv1alpha1.BentoRequest, dockerRegis
 }
 
 func isImageStoredInS3(bentoRequest *resourcesv1alpha1.BentoRequest) (storedInS3 bool) {
-	return bentoRequest.Annotations[commonconsts.KubeAnnotationImageStoredInS3] == commonconsts.KubeLabelValueTrue
+	return bentoRequest.Annotations[commonconsts.KubeAnnotationImageStoredInS3] == commonconsts.KubeLabelValueTrue && bentoRequest.Annotations["yatai.ai/bento-image-spec"] == "2"
```

Comment:
‌‌‌I have a small immature suggestion: can we control whether to use bento-image-builder only through `KubeAnnotationImageStoredInS3` on the bentocloud side, and set it to True by default? When encountering issues, remind users to set it to false in the UI.